### DISCUSSION TITLE
osquery windows logs parsing problem ?

### DISCUSSION THREAD

**Discussion by 0nnyx** [Security-Onion-Solutions/securityonion#2064] [Unsupported Versions] [answered]
I tried to move away my win10 hosts from winlogbeat for event monitoring which works fine to osquery using this advice: https://docs.securityonion.net/en/2.3/osquery.html#shipping-windows-eventlogs
When doing so, kibana logs all events with host.name=securityonion instead of the win10 hostname and EventData part is not parsed at all but remains a json blob as winlog.data, preventing any proper detection from playbook. Looks as if the parser does not support nested json nodes and parse only the 1st level.
Am wondering if I did anything wrong... Installed the launcher.msi, created a Fleet pack & a query polling SELECT * FROM windows_events; every 3 minutes as differential.

**Comment by defensivedepth** [comment, marked as answer]
What version of SO?

Also, please post a sanitized screencapture / snippet of what you are seeing in Kibana.

**Reply by 0nnyx** [reply]
Using 2.3.10 upgraded from 2.3.2 ISO version. Seems osquery can sometimes parse the nested json into temp.EvenData fields showing as unindexed, hence not able to be searched/matched and is unable to parse them otherwise.
Giving examples of sysmon process creation (but applies to other event types).

Below an example of parsed unindexed:

host.name | event.module | event.dataset | process.command_line
securityonion | sysmon | process_creation |  -

temp.EventData.CommandLine | "C:\WINDOWS\system32\taskmgr.exe" /4
  | temp.EventData.Company | Microsoft Corporation
  | temp.EventData.CurrentDirectory | C:\WINDOWS\System32\
  | temp.EventData.Description | Task Manager
  | temp.EventData.FileVersion | 10.0.19041.546 (WinBuild.160101.0800)

When using winlogbeat, fields are properly populated:

host.name | event.module | event.dataset | process.command_line
Robert | sysmon | process_creation | C:\WINDOWS\system32\vssvc.exe

process.command_line | C:\WINDOWS\system32\vssvc.exe
  | process.entity_id | {d089b9ea-fc9f-5fbc-c90b-000000003200}
  | process.executable | C:\Windows\System32\VSSVC.exe
  | process.parent.command_line | C:\WINDOWS\system32\services.exe


Below an example of sysmon event via osquery that's unparsable and only shown as winlog.data:

{"EventData":{"RuleName":"-","UtcTime":"2020-11-24 12:08:34.493","ProcessGuid":"{d089b9ea-f7c2-5fbc-6d0b-000000003200}","ProcessId":"11292","Image":"C:\\Windows\\System32\\svchost.exe","FileVersion":"10.0.19041.546 (WinBuild.160101.0800)","Description":"Host Process for Windows Services","Product":"Microsoft Windows Operating System","Company":"Microsoft Corporation","OriginalFileName":"svchost.exe","CommandLine":"C:\\WINDOWS\\system32\\svchost.exe -k netsvcs -s LxpSvc","CurrentDirectory":"C:\\WINDOWS\\system32\\","User":"AUTORITE NT\\Syst\xC3\xA8me","LogonGuid":"{d089b9a-e307-5fbb-e703-000000000000}","LogonId":"0x3e7","TerminalSessionId":"0","IntegrityLevel":"System","Hashes":"MD5=F586835082F632DC8D9404D83BC16316,SHA256=643EC58E82E0272C97C2A59F6020970D881AF19C0AD5029DB9C958C13B6558C7,IMPHASH=F9BBD96FAE53B7A31264A703CAFA0666","ParentProcessGuid":"{d089b9ea-e306-5fbb-0a00-000000003200}","ParentProcessId":"1020","ParentImage":"C:\\Windows\\System32\\services.exe","ParentCommandLine":"C:\\WINDOWS\\system32\\services.exe"}}

The win10 hostname is only visible with osquery.result.hostname field and not the host.name field.

**Reply by defensivedepth** [reply]
So the temp field parsing issue was [just fixed](https://github.com/Security-Onion-Solutions/securityonion/commit/426769588a6cb352571eb426d37697de89017d9d) and will be available in the next release, or you can do the following now to fix it:

Edit line 8 of ` /opt/so/saltstack/default/salt/elasticsearch/files/ingest/osquery.query_result` and change:

`{ "rename":      { "field": "temp.Data",               "target_field": "winlog.event_data",       "ignore_missing": true  } }, `

to

`{ "rename":      { "field": "temp.EventData",               "target_field": "winlog.event_data",       "ignore_missing": true  } },  `

Save it, and restart Elasticsearch:  `sudo so-elasticsearch-restart --force`

**Reply by 0nnyx** [reply]
If I understand right, the temp.EventData will be properly mapped after the fix. However, what about the occurences of EventData not parsed at all ? I guess it's still safer to run winlogbeat for global event shipping and use osquery for specific pack operations.

**Reply by defensivedepth** [reply]
Winlogbeat, Osquery & Wazuh all use the same Sysmon parsing config (once the logs are normalized). 

Once you implement the fix I mentioned, it should be parsed correctly. If not, let us know and we can get it fixed. The goal is that all three applications should fully parse Windows Eventlogs/Sysmon logs out of of the box.

**Reply by 0nnyx** [reply]
Tried with the change and looks better, will see if anything weird is still happening. Thanks
